### ISSUE TITLE
implicit_assignment_linter(allow_paren_print=TRUE) is wrong for multiple lints

### ISSUE THREAD

**Issue by MichaelChirico** [jimhester/lintr#3117]
Discovered incidentally by the AI reviewer of #3114:

```r
lint("
  (a <- 1L)
  mean(b <- 2L)
", implicit_assignment_linter(allow_paren_print=TRUE))
# <text>:3:8: warning: [implicit_assignment_linter] Call print() explicitly instead of relying on implicit printing behavior via '('.
#   mean(b <- 2L)
#        ^~~~~~~
# <text>:3:8: warning: [implicit_assignment_linter] Avoid implicit assignments in function calls. For example, instead of `if (x <- 1L) { ... }`, write `x <- 1L; if (x) { ... }`.
#   mean(b <- 2L)
#        ^~~~~~~
```

It gives a lint for the first expression with the wrong metadata (there should be no lint at all)
